component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/classification/hoeffdingtree/util/AdaptiveHoeffdingTreeModel.java

```diff
@@ -0,0 +1,240 @@
+     * @param splittingCriteria      Split criterion to use. 0:InfoGainSplitCriterion, 1:GiniSplitCriterion".
+     * @param allowableSplitError    Allowable error in split decision, values closer to 0 will take.
+     * @param breakTieThreshold      Threshold below which a split will be forced to break ties.
+     * @param binarySplitOption      Allow binary splits.
+     * @param disablePrePruning      Disable pre-pruning.
+     * @param leafpredictionStrategy Leaf prediction to use.
+     */
+    public void setConfigurations(int gracePeriod, int splittingCriteria,
+                                  double allowableSplitError, double breakTieThreshold,
+                                  boolean binarySplitOption, boolean disablePrePruning,
+                                  int leafpredictionStrategy) {
+        if (logger.isDebugEnabled()) {
+            logger.debug(String.format("Model [%s] is being configured with hyper-parameters.", this.modelName));
+        }
+        hoeffdingAdaptiveTree.gracePeriodOption.setValue(gracePeriod);
+        if (splittingCriteria == 0) {
+            hoeffdingAdaptiveTree.splitCriterionOption
+                    .setValueViaCLIString("InfoGainSplitCriterion");
+        } else {
+            hoeffdingAdaptiveTree.splitCriterionOption
+                    .setValueViaCLIString("GiniSplitCriterion");
+        }
+        hoeffdingAdaptiveTree.splitConfidenceOption.setValue(allowableSplitError);
+        hoeffdingAdaptiveTree.tieThresholdOption.setValue(breakTieThreshold);
+        hoeffdingAdaptiveTree.binarySplitsOption.setValue(binarySplitOption);
+        hoeffdingAdaptiveTree.noPrePruneOption.setValue(disablePrePruning);
+        hoeffdingAdaptiveTree.leafpredictionOption.setChosenIndex(leafpredictionStrategy);
+
+    }
+
+    /**
+     * @param cepEvent   event data
+     * @param classLabel class  label of the cepEvent
+     */
+    public void trainOnEvent(double[] cepEvent, String classLabel) {
+        cepEvent[noOfFeatures - 1] = addClass(classLabel);
+        Instance trainInstance = createMOAInstance(cepEvent);
+        trainInstance.setClassValue(cepEvent[noOfFeatures - 1]);
+        //training on the event instance
+        hoeffdingAdaptiveTree.trainOnInstanceImpl(trainInstance);
+    }
+
+    /**
+     * @param modelEvaluation Prequential Model Evaluator.
+     * @param cepEvent        event data
+     * @param classValue      class label of the cepEvent
+     * @return Prequential accuracy
+     */
+    public double evaluationTrainOnEvent(PrequentialModelEvaluation modelEvaluation,
+                                         double[] cepEvent, String classValue) {
+        int classIndex = cepEvent.length - 1;
+
+        //create instance with only the feature attributes
+        double[] test = Arrays.copyOfRange(cepEvent, 0, classIndex);
+        Instance testInstance = createMOAInstance(test);
+
+        double[] votes = hoeffdingAdaptiveTree.getVotesForInstance(testInstance);
+
```

Comment:
Don't have line spaces unnecessarily.

Reply:
Fixed in [#d4142ceef57683a0a3421cec3ee7fbd4cf114384](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/16/commits/d4142ceef57683a0a3421cec3ee7fbd4cf114384)